### DISCUSSION TITLE
Some questions about Pimcore and Workflow.

### DISCUSSION THREAD

**Discussion by alexmf23** [pimcore/pimcore#13267] [Q&A / Support]
Hello,
I have some questions about Pimcore and Workflow.

1.  In the View tab on Assets can I change the thumbnail to be displayed in higher quality ?
2. In assets when you upload a file with the same name it asks you if you want to keep both versions or overwrite them, but let's say we have a folder with multiple images in ASSETS and we reupload the same folder - it wont ask if you want to overwrite/keep both versions from the images inside the folder, it uploads and adding an increment _1 to those images.
3. Is it possible to send notification on asset upload ?
4. Is it possible to filter assets after workflow places ?

Thank you

**Comment by BlackbitDevs** [comment]
1. Currently the size is hard-coded. With higher quality you mean bigger image dimensions? Or what exactly?
2. I implemented this in #11020 and improved it in #13430 - maybe you can try the latter of it is already fixed there? If not please write info@blackbit.com, we can solve this for you (and for Pimcore community)
3. Not with standard Pimcore but with https://pimcore.com/en/developers/marketplace/blackbit_digital_commerce/pimcore-data-director_e103850 you can automatically execute any action when an element gets saved. In a tutorial video we have used this to [automatically assign an uploaded image to the corresponding product based on the asset's filename](https://www.youtube.com/watch?v=M1VVmpTjB_I&list=PL4-QRNfdsdKIfzQIP-c9hRruXf0r48fjt&index=8&t=2164s) - but of course you can also send a notification.
4. With default Pimcore no, with Data Director yes because the latter adds a grid operator to extract any information from assets so that you can show download URLs, asset properties, embedded metadata or your wanted workflow place.

**Reply by mattamon** [reply]
@alexmf23 
Regarding point 3, it is possible with standard pimcore.
You can always implement an Eventlistener and use the ```pimcore.asset.postAdd``` event to send a notification.

[Asset Events](https://github.com/pimcore/pimcore/blob/10.5/lib/Event/AssetEvents.php) -> All asset events.
[Eventmanager](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Event_API_and_Event_Manager.html) -> Implementing a listener.

[Notifications](https://pimcore.com/docs/pimcore/current/Development_Documentation/Tools_and_Features/Notifications.html) -> Standard pimcore notifications, but you can also send emails e.g.
It really depends on what kind of notification you would like.

4: what exactly do you mean by filter after workflow places?

**Reply by alexmf23** [reply]
Hello and thanks for the reply.
I have managed to increase the preview thumbnail of the assets (1)
I managed also to send notifications/emails using event listeners.

@mattamon 
At 4 actually i was looking for displaying the workflow place of each assets in the preview tab. 
Something like this: 
<img width="512" alt="Screenshot 2022-12-05 at 12 57 46" src="https://user-images.githubusercontent.com/63473281/205632003-7d3ddabd-8ece-4618-8dd7-ee54aae0ffdc.png">

I have managed to do it by creating a bundle and overriding getFolderContentPreviewAction() method like this:

```
            if ($asset->isAllowed('list')) {

                $assetid = $asset->getId();
                $assetWorkflowPlace = Element\WorkflowState::getByPrimary($assetid, 'asset', 'asset_workflow');
                $assetPlace = $assetWorkflowPlace ? $assetWorkflowPlace->getPlace() : null;
    
                $assets[] = [
                    'id' => $asset->getId(),
                    'type' => $asset->getType(),
                    'filename' => $asset->getFilename(),
                    'filenameDisplay' => htmlspecialchars($filenameDisplay),
                    'url' => $this->getThumbnailUrl($asset),
                    'idPath' => $data['idPath'] = Element\Service::getIdPath($asset),
                    'workflowState' => $assetPlace,
                ];
            }
```

**Reply by BlackbitDevs** [reply]
Ah, yes for preview mode this is a good solution for displaying the workflow place. Thought you wanted to display it on list / grid view.
